.github/workflows/test-e2e.yml

```diff
@@ -1,10 +1,11 @@
 name: test-e2e
 
 on:
-  workflow_call:
-    secrets:
-      KNUU_KUBECONFIG_FILE:
-        required: true
+  push:
+    branches:
+      - main
+    paths:
+      - "**.go"
```

Comment:
I think you may want to use something [like this](https://github.com/celestiaorg/celestia-app/blob/f9a7e04ef537ba5ec5af1e9a29055628778cc3f1/.github/workflows/lint.yml#L15-L22) so that the e2e-test runs if there are changes to `go.mod`